gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcCallContext.java

```diff
@@ -228,7 +250,8 @@ public GrpcCallContext withEndpointContext(EndpointContext endpointContext) {
         options,
         retrySettings,
         retryableCodes,
-        endpointContext);
+        endpointContext,
```

Comment:
Its unfortunate that we have to change all the places that use the constructor. Ideally, if we had a builder for `GrpcCallContext`, the code here would be simplified to `this.toBuilder().setEndpointContext(endpointContext).build()`, and we don't have to change the code here at all.

Reply:
Agreed. But I'd prefer to leave the refactoring with a builder out of this PR since it would look cleaner. I can open an issue for it. Let me know what you think.

Reply:
SG. Yes please create a separate issue and we can put it in our backlog.

Reply:
Thanks. Created #3681